pkg/kube/translation/knative_ingress.go

```diff
@@ -0,0 +1,149 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package translation
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	knativev1alpha1 "knative.dev/networking/pkg/apis/networking/v1alpha1"
+
+	"github.com/apache/apisix-ingress-controller/pkg/id"
+	"github.com/apache/apisix-ingress-controller/pkg/log"
+	apisixv1 "github.com/apache/apisix-ingress-controller/pkg/types/apisix/v1"
+)
+
+func (t *translator) translateKnativeIngressV1alpha1(ing *knativev1alpha1.Ingress) (*TranslateContext, error) {
+	ctx := &TranslateContext{
+		upstreamMap: make(map[string]struct{}),
+	}
+	plugins := t.translateAnnotations(ing.Annotations)
+
+	for i, rule := range ing.Spec.Rules {
+		hosts := rule.Hosts
+		if rule.HTTP == nil {
+			continue
+		}
+		for j, httpPath := range rule.HTTP.Paths {
+			var (
+				ups *apisixv1.Upstream
+				err error
+			)
+			knativeBackend := knativeSelectSplit(httpPath.Splits)
```

Comment:
Why just select one? Maybe others should be put to the traffic-split plugin.

Reply:
This is an initial version of Knative support, and `knativeSelectSplit` does not implement traffic split logic of KIngress right now. I plan to add KIngress features like traffic-split, append-headers, etc. in later PR. I can also add traffic-split feature in this PR.

Reply:
But it's so mysterious if we select one of them by the percent, I think just pick the first one is OK.

Reply:
OK, I'll just pick the first one.